gammapy/background/fov_cube.py

```diff
@@ -708,6 +710,58 @@ def make_spectrum(self, coord, ebounds=None):
 
         return spectrum
 
+    def make_roi_spectrum(self, offset, roi, ebounds=None):
+        """
+        Generate energy spectrum around a certain position in the FOV within an integration radius
+
+        Parameters
+        ----------
+        offset : `~astropy.units.Quantity`
+            Offset of the integration area
+        roi : `~astropy.units.Quantity`
+            Radius of the integration area
+        ebounds : `~gammapy.utils.energy.EnergyBounds`, optional
+            Energy binning for the spectrum
+
+        Returns
+        -------
+        spectrum : `~astropy.units.Quantity`
+            Energy spectrum
+        e_reco : `~astropy.units.Quantity`
+            Bounds of the reconstructed Energy
+        """
+
+        ebounds = self.energy_edges if ebounds is None else ebounds
+
+        xx, yy = np.meshgrid(self.coordx_edges[:].value, self.coordy_edges[:].value)
+        xx = Quantity(xx, self.coordx_edges[0].unit)
+        yy = Quantity(yy, self.coordx_edges[0].unit)
+        bc = zip(xx.ravel(), yy.ravel())
+        area = 0
+        spectrum = np.zeros(len(ebounds) - 1)
+        spectrum = Quantity(spectrum, self.data.unit)
+
+        for abc in bc:
+            vdist = (abc[0] ** 2 + abc[1] ** 2) ** 0.5
+
+        if np.abs(vdist - offset) <= roi:
+            area += 1.
+        coord = Angle([abc[0], abc[1]], abc[0].unit)
+        spectrum += self.make_spectrum(coord, ebounds=ebounds)
+
+        if area <= 0:
+            raise ValueError('No Off counts in the wished reagion')
+        area *= ((self.coordx_edges[1] - self.coordx_edges[0]) * (self.coordy_edges[1] - self.coordy_edges[0])).to(
+            'sr')
+        tarea = (math.pi * roi ** 2).to('sr')
+        delta_energy = self.energy_edges[1:] - self.energy_edges[:-1]
+
+        if len(delta_energy) != len(spectrum):
+            print("-> {} {} ".format(len(delta_energy), len(spectrum)))
```

Comment:
No print statements in Gammapy, please remove.
If this is useful info, then add it to the string passed to `ValueError`.

Reply:
yes, fixed!